plugins/userdata/JetUserData.cc

```diff
@@ -435,8 +436,40 @@ void JetUserData::fillJEC(pat::Jet& jet, const edm::Event& iEvent, const edm::Ev
 	jet.addUserFloat("L1OffJEC", jet.correctedJet("L1FastJet").pt() / jet.correctedJet("Uncorrected").pt());
 }
 
-void JetUserData::fillJER(pat::Jet& jet) const {
+void JetUserData::fillJER(pat::Jet& jet, const edm::Event& iEvent, const edm::EventSetup& iSetup) const {
```

Comment:
The problem I see with this implementation is that you are reading the jet resolution records for every jet!

Instead of passing `iEvent` and `iSetup` it would be better to pass it the `rho` value and the `JME::JetResolution` and `JME::JetResolutionScaleFactor` objects (references).

While the `rho` value needs to be read per event, the `JME::JetResolution` and `JME::JetResolutionScaleFactor` objects should be read in the constructor if `jerFromFile_ == true` or in the `produce()` method otherwise.
This will reduce the number of file open requests from `nEvents * nJets` to 1.